universal-login-react/src/ui/ConnectionFlow/ConnectWithEmoji.tsx

```diff
@@ -36,22 +36,31 @@ export const ConnectWithEmoji = ({name, sdk, onCancel, onConnect, walletService,
     }
   };
 
+  const TitledContent = ({title, children}: {title: string, children: ReactNode}) => <>
+    <h1 className="connect-emoji-title">{title}</h1>
+    <div className="connect-emoji-content">
+      <div className="connect-emoji-section">
+        {children}
+      </div>
+    </div>
+  </>;
+
   return (
     <div className="universal-login-emojis">
       <div className={getStyleForTopLevelComponent(className)}>
         <div className="connect-emoji">
-          <h1 className="connect-emoji-title">Confirmation</h1>
-          <div className="connect-emoji-content">
-            <div className="connect-emoji-section">
-              <p className="connect-emoji-text">Check the notification of another device controlling this account and type the emojis in this order.</p>
-              {!connectValues && !error && <Spinner className="spinner-center" />}
-              {connectValues && <div className="universal-login-emojis">
-                <EmojiPanel className="jarvis-styles" code={connectValues!.securityCode} />
-              </div>}
-              {error && `Error: ${error}`}
-              <button onClick={onCancelClick} className="connect-emoji-btn">Cancel Request</button>
-            </div>
-          </div>
+          {!error && <TitledContent title='Confirmation'>
+            <p className="connect-emoji-text">Check the notification of another device controlling this account and type the emojis in this order.</p>
+            {!connectValues && <Spinner className="spinner-center" />}
+            {connectValues && <div className="universal-login-emojis">
+              <EmojiPanel className="jarvis-styles" code={connectValues!.securityCode} />
+            </div>}
+            <button onClick={onCancelClick} className="connect-emoji-btn">Cancel Request</button>
+          </TitledContent>}
+          {error && <TitledContent title='Something went wrong.'>
```

Comment:
I would propose to extract more generic for example ErrorMessageScreen that we can use everywhere

Reply:
Then we don't need to extract TitledContent